galasa-extensions-parent/dev.galasa.extensions.common/src/main/java/dev/galasa/extensions/common/couchdb/CouchdbBaseValidator.java

```diff
@@ -0,0 +1,187 @@
+            StatusLine statusLine = response.getStatusLine();
+            int statusCode = statusLine.getStatusCode();
+
+            if (statusCode == HttpStatus.SC_OK) {
+                databaseExists = true;
+            } else if (statusCode != HttpStatus.SC_NOT_FOUND) {
+                String errorMessage = ERROR_FAILED_TO_VALIDATE_COUCHDB_DATABASE.getMessage(dbName, statusCode);
+                throw new CouchdbException(errorMessage);
+            }
+        } catch (IOException e) {
+            String errorMessage = ERROR_FAILED_TO_VALIDATE_COUCHDB_SERVER.getMessage(e.getMessage());
+            throw new CouchdbException(errorMessage, e);
+        }
+        return databaseExists;
+    }
+
+    /**
+     * Creates a new database with the given name, throwing a CouchdbException if
+     * the database could not be created.
+     *
+     * @param dbName the name of the database to be created
+     * @throws CouchdbException if there was a problem accessing the CouchDB server
+     *                          or creating the database
+     */
+    private synchronized void createDatabase(URI couchdbUri, String dbName) throws CouchdbException {
+        HttpPut httpPut = requestFactory.getHttpPutRequest(couchdbUri + "/" + dbName);
+        try (CloseableHttpResponse response = httpClient.execute(httpPut)) {
+            StatusLine statusLine = response.getStatusLine();
+            int statusCode = statusLine.getStatusCode();
+            if (statusCode != HttpStatus.SC_CREATED) {
+                EntityUtils.consumeQuietly(response.getEntity());
+
+                String errorMessage = ERROR_FAILED_TO_CREATE_COUCHDB_DATABASE.getMessage(dbName, statusCode);
+                throw new CouchdbException(errorMessage);
+
+            } else {
+                // The database was successfully created
+                EntityUtils.consumeQuietly(response.getEntity());
+            }
+        } catch (IOException e) {
+            String errorMessage = ERROR_FAILED_TO_CREATE_COUCHDB_DATABASE.getMessage(dbName, e.getMessage());
+            throw new CouchdbException(errorMessage, e);
+        }
+    }
+
+    /**
+     * Determines whether or not the given CouchDB version meets the minimum required CouchDB version
+     * for Galasa.
+     *
+     * @param actualVersion the version that the CouchDB server is running at
+     * @throws CouchdbException if the version of CouchDB is older than the minimum required version
+     */
+    private void checkVersion(String actualVersion) throws CouchdbException {
+        Pattern versionPattern = Pattern.compile("^(\\d+)\\.(\\d+)\\.(\\d+)$");
+        Matcher versionMatcher = versionPattern.matcher(actualVersion);
+
+        // Make sure the given version follows semantic versioning rules (i.e. major.minor.patch)
+        if (!versionMatcher.find()) {
+            String errorMessage = ERROR_INVALID_COUCHDB_VERSION.getMessage(actualVersion, COUCHDB_MIN_VERSION);
```

Comment:
This is a formatting issue with unexpected format returned from Couchdb, not a min check fail.

Reply:
Added a separate format-related error message